llvm/lib/Transforms/Instrumentation/SYCLITTAnnotations.cpp

```diff
@@ -0,0 +1,295 @@
+ *                                    __itt_atomic_mem_order_t mem_order);
+ * * * * * * * * * * *
+ * Atomic operation markup
+ *
+ * /param[in] object Memory location which is used in atomic operation
+ * /param[in] op_type Operation type
+ * /param[in] mem_order Memory ordering semantic
+ *
+ * void __itt_offload_atomic_op_finish(void* object,
+ *                                     __itt_atomic_mem_op_t op_type,
+ *                                     __itt_atomic_mem_order_t mem_order);
+ **/
+
+using namespace llvm;
+
+namespace {
+constexpr char SPIRV_PREFIX[] = "__spirv_";
+constexpr char SPIRV_CONTROL_BARRIER[] = "ControlBarrier";
+constexpr char SPIRV_GROUP_ALL[] = "GroupAll";
+constexpr char SPIRV_GROUP_ANY[] = "GroupAny";
+constexpr char SPIRV_GROUP_BROADCAST[] = "GroupBroadcast";
+constexpr char SPIRV_GROUP_IADD[] = "GroupIAdd";
+constexpr char SPIRV_GROUP_FADD[] = "GroupFAdd";
+constexpr char SPIRV_GROUP_FMIN[] = "GroupFMin";
+constexpr char SPIRV_GROUP_UMIN[] = "GroupUMin";
+constexpr char SPIRV_GROUP_SMIN[] = "GroupSMin";
+constexpr char SPIRV_GROUP_FMAX[] = "GroupFMax";
+constexpr char SPIRV_GROUP_UMAX[] = "GroupUMax";
+constexpr char SPIRV_GROUP_SMAX[] = "GroupSMax";
+constexpr char SPIRV_ATOMIC_INST[] = "Atomic";
+constexpr char SPIRV_ATOMIC_LOAD[] = "AtomicLoad";
+constexpr char SPIRV_ATOMIC_STORE[] = "AtomicStore";
+constexpr char ITT_ANNOTATION_WI_START[] = "__itt_offload_wi_start_wrapper";
+constexpr char ITT_ANNOTATION_WI_RESUME[] = "__itt_offload_wi_resume_wrapper";
+constexpr char ITT_ANNOTATION_WI_FINISH[] = "__itt_offload_wi_finish_wrapper";
+constexpr char ITT_ANNOTATION_WG_BARRIER[] = "__itt_offload_wg_barrier_wrapper";
+constexpr char ITT_ANNOTATION_ATOMIC_START[] = "__itt_offload_atomic_op_start";
+constexpr char ITT_ANNOTATION_ATOMIC_FINISH[] =
+    "__itt_offload_atomic_op_finish";
+
+// Wrapper for the pass to make it working with the old pass manager
+class SYCLITTAnnotationsLegacyPass : public ModulePass {
+public:
+  static char ID;
+  SYCLITTAnnotationsLegacyPass() : ModulePass(ID) {
+    initializeSYCLITTAnnotationsLegacyPassPass(
+        *PassRegistry::getPassRegistry());
+  }
+
+  // run the SYCLITTAnnotations pass on the specified module
+  bool runOnModule(Module &M) override {
+    ModuleAnalysisManager MAM;
+    auto PA = Impl.run(M, MAM);
+    return !PA.areAllPreserved();
+  }
+
+private:
+  SYCLITTAnnotationsPass Impl;
+};
+
+} // namespace
+
+char SYCLITTAnnotationsLegacyPass::ID = 0;
+INITIALIZE_PASS(SYCLITTAnnotationsLegacyPass, "SYCLITTAnnotations",
+                "Insert ITT annotations in SYCL code", false, false)
+
+// Public interface to the SYCLITTAnnotationsPass.
+ModulePass *llvm::createSYCLITTAnnotationsPass() {
+  return new SYCLITTAnnotationsLegacyPass();
+}
+
+namespace {
+
+// Check for calling convention of a function. If it's spir_kernel - consider
+// the function to be a SYCL kernel.
+bool isSyclKernel(Function &F) {
```

Comment:
```suggestion
// Check for calling convention of a function. If it's spir_kernel - consider
// the function to be a SYCL kernel.
bool isSPIRKernel(Function &F) {
```

Why is this pass SYCL specific? It should work for SPIR produced by other font-ends producing SPIR-V friendly LLVM IR. Right?
If you agree, please, replace SYCL with SPIR in file/pass names.

Reply:
You are right, thanks!